### DISCUSSION TITLE
Access static-build content from serverless function

### DISCUSSION THREAD

**Discussion by jakobrosenberg** [vercel/vercel#5265] [Help] [answered]
When `@vercel/static-build` generates content by running `package.json`'s `build` script, can I access the generated content from within a serverless function?

Or would I have to generate the same content twice? Once for the `@vercel/static-build` and once for `@vercel/node`

**Comment by asv7c2** [comment]
> When `@vercel/static-build` generates content by running `package.json`'s `build` script, can I access the generated content from within a serverless function?
> 
> Or would I have to generate the same content twice? Once for the `@vercel/static-build` and once for `@vercel/node`

Maybe this helps you.

[https://vercel.com/docs/runtimes#advanced-usage/advanced-node-js-usage/custom-build-step-for-node-js](https://vercel.com/docs/runtimes#advanced-usage/advanced-node-js-usage/custom-build-step-for-node-js)

**Reply by jakobrosenberg** [reply]
Thanks, but unless I'm missing something, that only explains how to build for node.

My question is about whether `@vercel/static-build` and `@vercel/node` can share the same build.

**Reply by asv7c2** [reply]
Main problem is that build script running in parallel with building serverless function. Vercel-build script is running before serverless function building. Try @vercel/static-build with vercel-build script and in serverless function try import or whatever you want with data from static build.

**Reply by jakobrosenberg** [reply]
Thanks @asv1. I've tried `includeFiles`, but get the error:
```
Error! Invalid vercel.json - `builds[1]` should NOT have additional property `includeFiles`. Please remove it.
```

I'll try use `functions` instead of `builds`, but as I recall, that didn't work either.

**Reply by jakobrosenberg** [reply]
@asv1 I'm still banging my head against the wall with this. Any chance you have a working example of a serverless function using the build from `static-build`?

**Reply by asv7c2** [reply]
> 
> 
> @asv1 I'm still banging my head against the wall with this. Any chance you have a working example of a serverless function using the build from `static-build`?

I can try. What kind of data you want share to serverless function? json, markdown or something else?

**Reply by jakobrosenberg** [reply]
Much appreciated! It can be anything really. For my use case, it's a SPA compiled to `bundle.js`. The bundle is served as a static asset, but also accessed by one of the serverless functions.

But if you're able to create a `.js` file with the `build` script in `package.json` and access it from a serverless function, then that's all I need.

**Comment by asv7c2** [comment, marked as answer]
I created [repository with example](https://github.com/asv1/tst).

static/build.js is called twice. one for build script and one for vercel-build in function 'func'.

**Reply by jakobrosenberg** [reply]
Thanks @asv1. I appreciate the effort, but this leaves me where I started - with a double build. If this doesn't affect pricing, I guess it's something I can live with.

**Reply by asv7c2** [reply]
You can try one hack. In vercel-build intead build script - run a script that waits (in loop) for a file bundle you want in public folder. When bundle will be created - read or require it. No double build than.

**Reply by jakobrosenberg** [reply]
Brilliant! That did the trick. Thanks so much.

I always assumed functions were built in separate containers.

**Reply by jakobrosenberg** [reply]
@asv1 The celebration was a little short lived. Your hack works when using the cli command `vercel`, but the same project deployed though Github didn't work. Do you know if there's a workaround?

**Reply by asv7c2** [reply]
build and vercel-build script is called? What it show in log?
